src/Analysis/Codelyzer.Analysis.CSharp/Handlers/InterfaceDeclarationHandler.cs

```diff
@@ -1,37 +1,66 @@
+using System.Collections.Generic;
 using Codelyzer.Analysis.Common;
 using Codelyzer.Analysis.Model;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 
 namespace Codelyzer.Analysis.CSharp.Handlers
 {
     public class InterfaceDeclarationHandler : UstNodeHandler
     {
+        // Key: interface full identifier, value: InterfaceDeclaration.
+        private Dictionary<string, InterfaceDeclaration> _interfaceDeclarationCache = new();
         private InterfaceDeclaration InterfaceDeclaration { get => (InterfaceDeclaration)UstNode; }
 
         public InterfaceDeclarationHandler(CodeContext context,
             InterfaceDeclarationSyntax syntaxNode)
             : base(context, syntaxNode, new InterfaceDeclaration())
         {
             var interfaceSymbol = SemanticHelper.GetDeclaredSymbol(syntaxNode, SemanticModel, OriginalSemanticModel);
-            InterfaceDeclaration.Identifier = syntaxNode.Identifier.ToString();
-
-
             if (interfaceSymbol != null)
             {
-                if (interfaceSymbol.BaseType != null)
+                var fullIdentifier = GetFullIdentifier(interfaceSymbol);
+                if (_interfaceDeclarationCache.ContainsKey(fullIdentifier))
                 {
-                    InterfaceDeclaration.BaseType = interfaceSymbol.BaseType.ToString();
-                    InterfaceDeclaration.BaseTypeOriginalDefinition = GetBaseTypOriginalDefinition(interfaceSymbol);
+                    UstNode = _interfaceDeclarationCache[fullIdentifier];
+                    return;
                 }
+                Set(InterfaceDeclaration, interfaceSymbol);
+                _interfaceDeclarationCache.Add(fullIdentifier, InterfaceDeclaration);
+            }
+        }
+        private void Set(InterfaceDeclaration interfaceDeclaration, INamedTypeSymbol interfaceSymbol)
+        {
+            interfaceDeclaration.Identifier = interfaceSymbol.Name;
```

Comment:
Same question here as above about parity of `Symbol.Name` and `SyntaxNode.Identifier`

Reply:
Same question. Symbol.Name assumes the semantic model was retrieved correctly. SyntaxNode.identifier is more generic.